web/__tests__/ProtectedRoute.test.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { render, cleanup } from 'react-testing-library';
+import { BrowserRouter } from 'react-router-dom';
+import { ProtectedRoute } from '../src/components/Auth';
+import { ACCESS_CONTROL } from '../src/constants/auth';
+
+afterEach(cleanup);
+
+describe('ProtectedRoute Component', () => {
+  class TestComponent extends Component {
+    render() {
+      return <h1>test content</h1>;
+    }
+  }
+
+  it('render content when user has correct role', () => {
+    const { container } = render(
+      <BrowserRouter>
+        <ProtectedRoute component={TestComponent} authCode={ACCESS_CONTROL.REQUESTER_ROLE} />
+      </BrowserRouter>
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  // TODO: Issue with testing on redirec: Maximum update depth exceeded.
```

Comment:
maybe mock out redirect to just to return a react component you can test for?

Reply:
mocking not working, will try later